Source/CarthageKit/Project.swift

```diff
@@ -1228,63 +1247,63 @@ internal func relativeLinkDestination(for dependency: Dependency, subdirectory:
+public func cloneOrFetch(dependency: Dependency, preferHTTPS: Bool,
+                         destinationURL: URL = carthageDependencyRepositoriesURL,
+                         commitish: String? = nil) -> SignalProducer<(ProjectEvent?, URL), CarthageError> {
 	let fileManager = FileManager.default
 	let repositoryURL = repositoryFileURL(for: dependency, baseURL: destinationURL)
 
 	return SignalProducer.attempt { () -> Result<GitURL, CarthageError> in
-			do {
-				try fileManager.createDirectory(at: destinationURL, withIntermediateDirectories: true)
-			} catch let error as NSError {
-				return .failure(.writeFailed(destinationURL, error))
-			}
-
-			return .success(dependency.gitURL(preferHTTPS: preferHTTPS)!)
+		do {
+			try fileManager.createDirectory(at: destinationURL, withIntermediateDirectories: true)
+		} catch let error as NSError {
+			return .failure(.writeFailed(destinationURL, error))
 		}
-		.flatMap(.merge) { remoteURL -> SignalProducer<(ProjectEvent?, URL), CarthageError> in
-			return isGitRepository(repositoryURL)
-				.flatMap(.merge) { isRepository -> SignalProducer<(ProjectEvent?, URL), CarthageError> in
-					if isRepository {
-						let fetchProducer: () -> SignalProducer<(ProjectEvent?, URL), CarthageError> = {
-							guard FetchCache.needsFetch(forURL: remoteURL) else {
-								return SignalProducer(value: (nil, repositoryURL))
-							}
 
-							return SignalProducer(value: (.fetching(dependency), repositoryURL))
-								.concat(
-									fetchRepository(repositoryURL, remoteURL: remoteURL, refspec: "+refs/heads/*:refs/heads/*")
-										.then(SignalProducer<(ProjectEvent?, URL), CarthageError>.empty)
-								)
-						}
-
-						// If we've already cloned the repo, check for the revision, possibly skipping an unnecessary fetch
-						if let commitish = commitish {
-							return SignalProducer.zip(
-									branchExistsInRepository(repositoryURL, pattern: commitish),
-									commitExistsInRepository(repositoryURL, revision: commitish)
-								)
-								.flatMap(.concat) { branchExists, commitExists -> SignalProducer<(ProjectEvent?, URL), CarthageError> in
-									// If the given commitish is a branch, we should fetch.
-									if branchExists || !commitExists {
-										return fetchProducer()
-									} else {
-										return SignalProducer(value: (nil, repositoryURL))
-									}
-								}
-						} else {
-							return fetchProducer()
-						}
-					} else {
-						// Either the directory didn't exist or it did but wasn't a git repository
-						// (Could happen if the process is killed during a previous directory creation)
-						// So we remove it, then clone
-						_ = try? fileManager.removeItem(at: repositoryURL)
-						return SignalProducer(value: (.cloning(dependency), repositoryURL))
+		return .success(dependency.gitURL(preferHTTPS: preferHTTPS)!)
+	}
+	.flatMap(.merge) { remoteURL -> SignalProducer<(ProjectEvent?, URL), CarthageError> in
```

Comment:
Again, I don't see how this can be further indented, see above comment with `}`.